### PR TITLE
Add CLI commands for repository enrollment management

### LINKED ISSUE #695: Add dedicated enable/disable commands for repository enrollment
## Desired behavior

Add dedicated commands to enable/disable repository enrollment after installation.

**Enable command:**
```bash
fullsend admin enable repos <org> <repo-1> <repo-2> ...
fullsend admin enable repos <org> --all
```

- When given repository names, enables those repositories for fullsend enrollment
- When given `--all`, enables all repositories (excluding `.fullsend`)
- If `--all` is specified, positional repository arguments are ignored

**Disable command:**
```bash
fullsend admin disable repos <org> <repo-1> <repo-2> ...
fullsend admin disable repos <org> --all
```

- When given repository names, disables those repositories
- When given `--all`, disables all repositories
- If `--all` is specified, positional repository arguments are ignored

**Note:** The commands use a `repos` subcommand to allow future extensibility for enabling/disabling other configuration options (e.g., `fullsend admin enable auto-enrollment`, `fullsend admin disable auto-enrollment`).

## Context

The current enrollment model requires editing `config.yaml` directly or re-running `fullsend admin install`. Dedicated commands provide a clearer workflow and allow users to adjust enrollment over time.

These commands will:
1. Read `.fullsend/config.yaml` from the repository
2. Update the specified repos to `enabled: true` or `enabled: false`
3. Commit and push changes
4. Trigger the `repo-maintenance.yml` workflow to reconcile enrollment state

## Implementation notes

- Commands should validate that `.fullsend` repository exists
- Should validate repository names match actual org repos
- Should respect exclusion patterns (always exclude `.fullsend` itself)
- Should provide clear feedback about what changed

## Related

Extracted from #495
Required for #692 (auto-enrollment feature)

### PR DESCRIPTION
**Pull request by ggallen**
**Summary**

Implements #695 by adding dedicated CLI commands to manage repository enrollment state in the `.fullsend` config repository.

**Changes**

*   **New command structure**: `fullsend admin repos enable <org> [repo...]` and `fullsend admin repos disable <org> [repo...]`
*   **\--all flag**: Enable/disable all discovered repositories at once
*   **Flag behavior**: When `--all` is set, positional repository arguments are silently ignored
*   **Validation**: Verifies repository existence before modifying config.yaml
*   **Workflow trigger**: Automatically triggers repo-maintenance.yml after config updates

**Test plan**

*    Unit tests verify command structure and flag handling
*    Flag behavior tested: `--all` ignores positional arguments (both enable and disable)
*    Error handling tested for missing org, conflicting flags, and invalid inputs
*    Integration with forge.Client abstraction layer verified

**Related**

Closes #695

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by rh-hemartin — CHANGES_REQUESTED** [review]
Weren't we going to use `fullsend admin enable repos`?

https://github.com/fullsend-ai/fullsend/issues/495#issuecomment-4359101521

**Comment by ggallen** [comment]
> Weren't we going to use `fullsend admin enable repos`?
> 
Yeah, I think this is backward.

**Comment by ggallen** [comment]
@rh-hemartin Your review feedback has been addressed:

1. ✅ **Command structure corrected**: Changed from `fullsend admin repos enable/disable` to `fullsend admin enable/disable repos` as specified in issue #495. This provides better extensibility for future commands like `fullsend admin enable auto-enrollment`.

2. ✅ **Method naming updated**: Renamed `runReposEnable` → `runEnableRepos` and `runReposDisable` → `runDisableRepos` for consistency with the command structure.

3. ✅ **Additional improvements** (from bot review):
   - Restored `--repo` flag test assertion in `TestInstallCmd_Flags`
   - Added confirmation prompt for `disable --all` operations with `--yolo` flag to skip (consistent with uninstall command)

All tests pass. Ready for re-review.

File: internal/cli/admin.go [hunk 1]
```diff
@@ -1006,6 +1009,369 @@ func promptDispatchToken(ctx context.Context, client forge.Client, printer *ui.P
...
+	} else {
+		// Validate provided repo names.
+		printer.StepStart("Validating repository names")
+		for _, repo := range repos {
+			if repo == forge.ConfigRepoName {
+				printer.StepFail("Cannot disable .fullsend repository")
+				return fmt.Errorf("cannot disable .fullsend repository itself")
+			}
+			// Check if repo exists in org.
+			_, err := client.GetRepo(ctx, org, repo)
```
Comment (ralphbean) [inline internal/cli/admin.go]: **[moderate]** The disable path validates repo existence via `GetRepo`, which blocks disabling repos that were deleted from the org but still have entries in `config.yaml`. The user gets "repository X not found" with no recourse except `disable --all`.

For the disable path specifically, consider falling back to a config-entry lookup when `GetRepo` returns not-found, or skipping the org-level existence check entirely (the enable path correctly validates against GitHub since you want to confirm the repo exists before enrolling, but disable doesn't need this guarantee).

File: internal/cli/admin.go [hunk 2]
```diff
@@ -1006,6 +1009,369 @@ func promptDispatchToken(ctx context.Context, client forge.Client, printer *ui.P
...
+		Use:   "disable",
+		Short: "Disable fullsend features",
+		Long:  "Commands for disabling fullsend features such as repository enrollment.",
+	}
+	cmd.AddCommand(newDisableReposCmd())
+	return cmd
+}
+
+// reposRunFunc is the signature for repo enable/disable operations.
+type reposRunFunc func(ctx context.Context, client forge.Client, printer *ui.Printer, org string, repos []string, all bool, yolo bool) error
```
Comment (ralphbean) [inline internal/cli/admin.go:1034]: **[minor, deferred]** The `yolo` parameter in `reposRunFunc` is accepted by both commands via the shared type signature, but `runEnableRepos` never reads it — the `--yolo` flag silently does nothing on enable. Consider either not exposing `--yolo` on the enable command (split the flag registration out of `newReposSubcommand`) or adding a confirmation prompt for `enable --all` for consistency.

File: docs/guides/admin/installation.md [hunk 3]
```diff
@@ -187,7 +187,50 @@ After install completes, the installer dispatches a workflow that creates an enr
...
+
+```bash
+fullsend admin disable repos "$ORG_NAME" --all
+```
+
+The disable command:
+- Updates `config.yaml` to mark repositories as disabled
+- Triggers the `repo-maintenance` workflow to create unenrollment PRs
+- Validates that repositories exist in the organization
+- Does not delete existing shim workflows (merge the unenrollment PR to remove them)
```
Comment (ralphbean) [inline docs/guides/admin/installation.md:237]: **[minor, deferred]** The docs for `disable --all` don't mention the confirmation prompt (user must type the org name) or the `--yolo` flag to skip it. An admin running `disable --all` in a script would hit an unexpected interactive prompt.

**Review by ralphbean — CHANGES_REQUESTED** [review]
Good implementation overall — the shared `newReposSubcommand` factory, extracted `loadRepoConfig`/`saveRepoConfig` helpers, and comprehensive test suite are well done. The intentional asymmetry between enable --all (discovers org repos) and disable --all (iterates config entries) is a smart design choice.

Two changes requested and two notes for consideration.

---

**[moderate] `TestAdminCommand_HasSubcommands` not updated for new subcommands**

`TestAdminCommand_HasSubcommands` (admin_test.go:17-25) asserts that `admin` has `install`, `uninstall`, and `analyze` subcommands, but does not check for `enable` and `disable`. If someone removes the `cmd.AddCommand(newEnableCmd())` line, no test catches it. Please add:

```go
assert.True(t, names["enable"], "expected enable subcommand")
assert.True(t, names["disable"], "expected disable subcommand")
```

**Review by ggallen — COMMENTED** [review]
**Thread 1 (disable validation):** Fixed in 6be0cd07. The disable operation now validates against config instead of GitHub, allowing cleanup of repos that have been deleted from the org. Updated test  to verify this behavior.

**Thread 2 (unused yolo parameter):** Acknowledged as minor/deferred. The yolo flag remains in the shared function signature for consistency, even though enable doesn't use it. This could be refactored in a future PR if needed.

**Thread 3 (documentation):** Fixed in 6be0cd07. Added documentation for the  confirmation prompt and the  flag to skip it for scripted usage.

**Review by ggallen — COMMENTED** [review]
Thread 1 (disable validation): Fixed in 6be0cd07. The disable operation now validates against config instead of GitHub, allowing cleanup of repos that have been deleted from the org. Updated test TestRunDisableRepos_AllowsRepoNotInConfig to verify this behavior.

Thread 2 (unused yolo parameter): Acknowledged as minor/deferred. The yolo flag remains in the shared function signature for consistency, even though enable does not use it. This could be refactored in a future PR if needed.

Thread 3 (documentation): Fixed in 6be0cd07. Added documentation for the --all confirmation prompt and the --yolo flag to skip it for scripted usage.

File: internal/cli/admin.go (hunk 1 above)
Comment (ggallen) [reply internal/cli/admin.go]: Fixed in 6be0cd07. Removed GitHub validation from the disable operation - it now validates against config instead. This allows cleanup of repos that have been deleted from the org but still have config entries. Updated the test to verify this behavior: `TestRunDisableRepos_ErrorWhenRepoNotFound` renamed to `TestRunDisableRepos_AllowsRepoNotInConfig`.

File: internal/cli/admin.go (hunk 2 above)
Comment (ggallen) [reply internal/cli/admin.go:1034]: Acknowledged as minor/deferred. The yolo flag remains in the shared `reposRunFunc` signature for consistency, even though `runEnableRepos` does not currently use it. This could be refactored in a future PR if needed (either remove the flag from enable, or add a confirmation prompt to enable --all for symmetry).

File: docs/guides/admin/installation.md (hunk 3 above)
Comment (ggallen) [reply docs/guides/admin/installation.md:237]: Fixed in 6be0cd07. Added documentation for the `--all` confirmation prompt and the `--yolo` flag. The docs now explain that `disable --all` requires typing the organization name for confirmation, and shows how to use `--yolo` to skip the prompt for scripted usage.

File: internal/cli/admin.go [hunk 4]
```diff
@@ -1006,6 +1009,370 @@ func promptDispatchToken(ctx context.Context, client forge.Client, printer *ui.P
...
+		// Unlike enable, disable is cleanup and must handle repos deleted from GitHub.
+		printer.StepStart("Validating repository names")
+		for _, repo := range repos {
+			if repo == forge.ConfigRepoName {
+				printer.StepFail("Cannot disable .fullsend repository")
+				return fmt.Errorf("cannot disable .fullsend repository itself")
+			}
+			// Check if repo exists in config (don't require GitHub existence for cleanup).
+			if _, exists := cfg.Repos[repo]; !exists {
+				printer.StepWarn(fmt.Sprintf("Repository %s not in config (skipping)", repo))
```
Comment (ralphbean) [inline internal/cli/admin.go:1280]: **[moderate, deferred]** The warning says "skipping" but the repo is still appended to `reposToDisable` (line 1259 assigns `repos` wholesale). The update loop at line 1272 safely handles this (checks `exists`), so it's functionally correct, but the control flow is misleading — the message implies the repo is excluded from processing when it isn't.

Consider adding `continue` after the warning to actually skip, or filtering `reposToDisable` to only include repos present in config.

File: internal/cli/admin.go [hunk 5]
```diff
@@ -1006,6 +1009,370 @@ func promptDispatchToken(ctx context.Context, client forge.Client, printer *ui.P
...
+	if err := client.CreateOrUpdateFile(ctx, org, forge.ConfigRepoName, "config.yaml", commitMsg, updatedConfigData); err != nil {
+		printer.StepFail("Failed to commit changes")
+		printer.StepInfo("Hint: verify your token has 'repo' scope with: gh auth refresh -s repo")
+		return fmt.Errorf("committing config.yaml: %w", err)
+	}
+	printer.StepDone("Changes committed to .fullsend")
+
+	// Trigger repo-maintenance workflow.
+	printer.StepStart("Triggering repo-maintenance workflow")
+	if err := client.DispatchWorkflow(ctx, org, forge.ConfigRepoName, "repo-maintenance.yml", "main", nil); err != nil {
```
Comment (ralphbean) [inline internal/cli/admin.go:1390]: **[moderate, deferred]** Several error/edge-case paths in the enable/disable flow lack test coverage:

1. `ListOrgRepos` failure when `enable --all` is used
2. `CreateOrUpdateFile` failure during `saveRepoConfig`
3. This line — `DispatchWorkflow` failure is intentionally treated as a non-fatal warning (returns `nil`), but no test verifies this important business behavior
4. The `yolo=false` confirmation prompt path (`fmt.Scanln`) — all `disable --all` tests pass `yolo=true`

None of these are blocking, but worth adding in a follow-up to prevent regressions.

File: docs/guides/admin/installation.md [hunk 6]
```diff
@@ -187,7 +187,56 @@ After install completes, the installer dispatches a workflow that creates an enr
...
+The `--all` flag prompts for confirmation by requiring you to type the organization name. To skip the prompt (e.g., in scripts):
+
+```bash
+fullsend admin disable repos "$ORG_NAME" --all --yolo
+```
+
+The disable command:
+- Updates `config.yaml` to mark repositories as disabled
+- Triggers the `repo-maintenance` workflow to create unenrollment PRs
+- Validates repository names against the config (not GitHub) to allow cleanup of deleted repos
```
Comment (ralphbean) [inline docs/guides/admin/installation.md]: **[minor, deferred]** The word "validates" is slightly misleading here. The implementation warns about repo names not found in config but does not reject them — the command still succeeds. Consider rewording to something like:

> Warns (but does not reject) repository names not found in the config, allowing safe cleanup of deleted repos

File: internal/cli/admin.go [hunk 7]
```diff
@@ -1006,6 +1009,370 @@ func promptDispatchToken(ctx context.Context, client forge.Client, printer *ui.P
...
+			var repos []string
+			if all {
+				// Ignore positional args; repos will be discovered from org
+				repos = nil
+			} else {
+				hasRepos := len(args) > 1
+				if !hasRepos {
+					return fmt.Errorf("must specify repository names or use --all flag")
+				}
+				repos = args[1:]
```
Comment (ralphbean) [inline internal/cli/admin.go:1064]: **[nit, deferred]** No deduplication of positional repo arguments. Running `fullsend admin enable repos myorg web-app web-app` causes redundant `GetRepo` API calls during validation. The update loop handles it correctly (second pass is a no-op), but a `slices.Compact(slices.Sorted(...))` or similar dedup here would avoid wasted API calls.

**Review by ralphbean — APPROVED** [review]
Good implementation overall. The shared `newReposSubcommand` factory, extracted helpers, intentional enable/disable asymmetry, and comprehensive test suite are well done. A few notes for future consideration below — nothing blocking.

**Review by ggallen — COMMENTED** [review]
Review feedback addressed in ef044849:

1. **Disable repos skipping behavior**: Fixed the misleading "skipping" message. The code now actually skips repos not in config by adding a `continue` statement and only appending repos that exist in config to `reposToDisable`.

2. **Documentation clarity**: Changed the wording from "validates repository names against the config" to "warns (but does not reject) repository names not found in the config" to accurately describe the behavior.

The changes ensure that when disabling repos, the command correctly handles repos that have been deleted from GitHub while still being present in config (cleanup scenario), and repos that never existed in config are properly skipped with a warning.

File: internal/cli/admin.go (hunk 1 above)
Comment (ggallen) [reply internal/cli/admin.go]: Fixed in ef044849. The disable path now validates repository names against the config (not GitHub), allowing cleanup of repos that have been deleted from the org. When a repo is not found in config, it emits a warning and skips the repo.

File: internal/cli/admin.go (hunk 4 above)
Comment (ggallen) [reply internal/cli/admin.go:1280]: Fixed in ef044849. Added `continue` statement after the warning, so repos not in config are actually skipped instead of being added to `reposToDisable`. The code now builds the list incrementally by appending only repos that exist in config.

File: docs/guides/admin/installation.md (hunk 6 above)
Comment (ggallen) [reply docs/guides/admin/installation.md]: Fixed in ef044849. Changed the wording from "Validates repository names against the config (not GitHub) to allow cleanup of deleted repos" to "Warns (but does not reject) repository names not found in the config, allowing safe cleanup of deleted repos" to accurately describe the non-blocking behavior.

**Review by rh-hemartin — APPROVED** [review]
